File: solutions/_242_valid_anagram/solution.js
/**
 * @param {string} s
 * @param {string} t
 * @return {boolean}
 */
var isAnagram = function (s, t) {
  if (s.length !== t.length) return false;

  const ctr = new Array(26).fill(0);

  for (let i = 0; i < s.length; i++) {
    ++ctr[s.charCodeAt(i) - "a".charCodeAt(0)];
    --ctr[t.charCodeAt(i) - "a".charCodeAt(0)];
  }

  return ctr.every((count) => count === 0);
};

module.exports = { isAnagram };

/*
https://leetcode.com/problems/valid-anagram/
*/

File: solutions/_242_valid_anagram/solution.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        length_s = len(s)
        length_t = len(t)
        if length_s != length_t:
            return False

        ctr = [0] * 26

        for i in range(length_s):
            ctr[ord(s[i]) - 97] += 1
            ctr[ord(t[i]) - 97] -= 1

        for i in range(26):
            if ctr[i] != 0:
                return False

        return True


# https://leetcode.com/problems/valid-anagram/
